src/core_plugins/kibana/index.js

```diff
@@ -84,7 +90,26 @@ module.exports = function (kibana) {
       ingest(server);
       search(server);
       settings(server);
+
+      registerCoreTranslations();
     }
   });
 
 };
+
+function registerCoreTranslations()
+{
+  const rootDir = kibanaPackage.__dirname;
+
+  //Add translation dirs for the core plugins here
+  const corePluginTranslationDirs = [rootDir + '/src/ui/i18n'];
+
+  return Promise.map(corePluginTranslationDirs, (dir) => {
+    readdir(dir).then((dirListing) => {
+      Promise.map(dirListing, (listing) => {
+        const fullFilePath = dir + '/' + listing;
+        i18nPlugin.registerTranslations(fullFilePath);
+      });
+    });
+  });
```

Comment:
Since we're only going to have english translations in Kibana core for the foreseeable future, let's avoid this complexity and just register the one en.json file directly.